airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2146,22 +2149,28 @@ def create_async_retriever(
             urls_extractor=urls_extractor,
         )
 
-        return AsyncRetriever(
+        async_job_partition_router = AsyncJobPartitionRouter(
             job_orchestrator_factory=lambda stream_slices: AsyncJobOrchestrator(
                 job_repository,
                 stream_slices,
-                JobTracker(
-                    1
-                ),  # FIXME eventually make the number of concurrent jobs in the API configurable. Until then, we limit to 1
+                JobTracker(1),
+                # FIXME eventually make the number of concurrent jobs in the API configurable. Until then, we limit to 1
```

Comment:
[outside of the scope of this PR but important] This is very interesting! I don't remember this very well but it seems that we can only have one concurrent job on the API side at the same time until we tackle this `FIXME`. 

A couple of notes/questions:
* [source-salesforce](https://github.com/airbytehq/airbyte/blob/2370c070f42109a7c9c055629b468835909f97d9/airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py#L77) is using 5 concurrent jobs and it seems to be working well so we have a case where the limit is not 1 that has been tested in prod
* It feels like the instantiation of this `JobTracker` is wrong. All instances of `AsyncJobOrchestrator` should probably share the same instance as the limit on the API side is usually not per stream. It is done properly in source-salesforce and probably is not an issue source-sendgrid as there is only one stream with a job
* @artem1205 do you know how many concurrent job we can have for the Amazon Ads API and for the Amazon Seller Partner API? If there benefit in making it configurable now?

Reply:
i'll write up a ticket for this since it's something we'll want configurable at least at the manifest level (or customer config level but that we can decide later).